components/eamxx/src/physics/mam4/aerosol_microphysics.cpp

```diff
@@ -0,0 +1,171 @@
+#include <physics/mam4/aerosol_microphysics.hpp>
+#include <share/property_checks/field_lower_bound_check.hpp>
+#include <share/property_checks/field_within_interval_check.hpp>
+
+#include "scream_config.h" // for SCREAM_CIME_BUILD
+
+#include <ekat/ekat_assert.hpp>
+
+namespace scream
+{
+
+MAM4AerosolMicrophysics::MAM4AerosolMicrophysics(
+    const ekat::Comm& comm,
+    const ekat::ParameterList& params)
+  : AtmosphereProcess(comm, params),
+    nucleation_(nullptr) {
+}
+
+AtmosphereProcessType MAM4AerosolMicrophysics::type() const {
+  return AtmosphereProcessType::Physics;
+}
+
+std::string MAM4AerosolMicrophysics::name() const {
+  return "MAM4AerosolMicrophysics";
+}
+
+void MAM4AerosolMicrophysics::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+
+  // The units of mixing ratio q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto q_unit = kg/kg;
+  q_unit.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  grid_ = grids_manager->get_grid("Physics");
+  const auto& grid_name = grid_->name();
+
+  ncol_ = grid_->get_num_local_dofs(); // Number of columns on this rank
+  nlev_ = grid_->get_num_vertical_levels(); // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 2D (1d horiz X 1d vertical) variable
+  FieldLayout scalar2d_layout_col{ {COL}, {ncol_} };
+
+  // Layout for 3D (2d horiz X 1d vertical) variables
+  FieldLayout scalar3d_layout_mid{ {COL, LEV}, {ncol_, nlev_} };
+
+  // Define fields needed in mam4xx.
+  const auto m2 = m*m;
+  const auto s2 = s*s;
+
+  // atmospheric quantities
+  add_field<Required>("T_mid", scalar3d_layout_mid, K, grid_name);
+  add_field<Required>("p_mid", scalar3d_layout_mid, Pa, grid_name); // total pressure
+  add_field<Required>("qv", scalar3d_layout_mid, q_unit, grid_name, "tracers");
+  add_field<Required>("pbl_height", scalar2d_layout_col, m, grid_name);
+  add_field<Required>("pseudo_density", scalar3d_layout_mid, q_unit, grid_name); // pdel
+  add_field<Required>("cldfrac_tot", scalar3d_layout_mid, nondim, grid_name);
+
+  // tracer group (stores all aerosol prognostics)
+  add_group<Updated>("tracers", grid_name, Bundling::Required);
```

Comment:
I'm assuming mam does not update all tracers, but just the aerosols ones? At some point, we may want to restrict it to those only, and see how robust it the FieldManager ability to rearrange fields inside groups (which aims at keeping as many as possible subgroups contiguous).

Reply:
Your assumption is mostly correct--we'll also need to update some gas tracers that interact with aerosols. I agree it would be nice if we can devise a way to restrict updates to tracer subgroups.